flocker/acceptance/endtoend/test_leases.py

```diff
@@ -0,0 +1,229 @@
+            return d
+        acquiring_lease = getting_dataset_state.addCallback(acquire_lease)
+
+        def start_http_container(ignored):
+            """
+            Launch data HTTP container and make POST requests
+            to it in a looping call every second.
+            return looping call deferred
+            """
+            [dataset] = datasets
+            script = SCRIPTS.child("datahttp.py")
+            script_arguments = [u"/data"]
+            docker_arguments = {
+                "host_config": create_host_config(
+                    binds=["{}:/data".format(dataset.path.path)],
+                    port_bindings={http_port: http_port}),
+                "ports": [http_port],
+                "volumes": [u"/data"]}
+            container = client.create_container(
+                "python:2.7-slim",
+                ["python", "-c", script.getContent()] + list(script_arguments),
+                **docker_arguments)
+            container_id = container["Id"]
+            containers.insert(0, container_id)
+            client.start(container=container_id)
+            self.addCleanup(client.remove_container, container_id, force=True)
+        starting_http_container = acquiring_lease.addCallback(
+            start_http_container
+        )
+
+        def write_data(ignored):
+            data = random_name(self).encode("utf-8")
+            d = post_http_server(
+                self, cluster.nodes[0].public_address, http_port,
+                {"data": data}
+            )
+            d.addCallback(
+                lambda _: assert_http_server(
+                    self, cluster.nodes[0].public_address,
+                    http_port, expected_response=data
+                )
+            )
+            return d
+        writing_data = starting_http_container.addCallback(write_data)
+
+        def stop_container(ignored):
+            """
+            This ensures Docker hasn't got a lock on the volume that might
+            prevent it being moved separate to the lock held by the lease.
+            """
+            [container_id] = containers
+            client.stop(container_id)
+        stopping_container = writing_data.addCallback(stop_container)
+
+        def perform_operation(ignored):
+            return operation(
+                dataset_id=dataset_id, **additional_kwargs
+            )
+        performing_operation = stopping_container.addCallback(
+            perform_operation
+        )
+
+        def wait_for_unexpected_umount(ignored):
+            """
+            If the dataset agent is broken and not respecting leases then we
+            expect that 10 seconds is long enough for it to begin performing
+            the requested operation.
+            And the first step will always be to unmount the filesystem which
+            should happen quickly since there are no open files and no Docker
+            bind mounts to the filesystem.
+            """
```

Comment:
I'm sorry, I'm probably missing something obvious, but I don't quite understand this. 

Does this mean that if the dataset agent is broken then it will unbreak after 10 seconds? Or does it mean that if it's not working after 10 seconds then it's definitely broken? 

Also, I don't quite see how the second sentence relates to the first sentence. It says "And the first step" but I can't tell what it's the first step of.